backend/src/ncpdpScriptBuilder/buildScript.v2017071.js

```diff
@@ -46,14 +54,130 @@ export default function buildRxStatus(newOrder) {
       }
     ]
   };
-  const options = {
-    ignoreAttributes: false,
-    attributeNamePrefix: '@@',
-    format: true,
-    oneListGroup: 'true'
-  };
-  const builder = new XMLBuilder(options);
-  var RxStatus = builder.build(rxStatus);
 
-  return RxStatus;
+  const builder = new XMLBuilder(XML_BUILDER_OPTIONS);
+  return builder.build(rxStatus);
 }
+
+export const buildRxFill = newRx => {
+  const { Message } = JSON.parse(newRx.serializedJSON);
+  const { Header, Body } = Message;
+  const time = new Date();
+  const message = {
+    Message: {
+      Header: [
+        {
+          To: {
+            '#text': Header.From._,
+            '@@Qualifier': Header.From.$.Qualifier
+          }
+        },
+        {
+          From: {
+            '#text': Header.To._,
+            '@@Qualifier': Header.To.$.Qualifier
+          }
+        },
+        { MessageID: uuidv4() },
+        { RelatesToMessageID: Header.MessageID },
+        { SentTime: time.toISOString() },
+        { PrescriberOrderNumber: Header.PrescriberOrderNumber }
+      ],
+      Body: [
+        {
+          RxFill: {
+            FillStatus: {
+              Dispensed: {
+                Note: PICKED_UP
+              }
+            },
+            Patient: {
```

Comment:
Can you just put: `Patient: Body.NewRx.Patient` ?

Reply:
actually, yeah, that's possible